public/app/plugins/datasource/tempo/QueryEditor/ServiceGraphSection.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import LokiDatasource from '../../loki/datasource';
+import useAsync from 'react-use/lib/useAsync';
+import { getDS } from './utils';
+import { InlineField, InlineFieldRow } from '@grafana/ui';
+import { AdHocVariableFilter } from '../../../../features/variables/types';
+import { TempoQuery } from '../datasource';
+import { AdHocFilter } from '../../../../features/variables/adhoc/picker/AdHocFilter';
+
+export function ServiceGraphSection({
+  graphDatasourceUid,
+  query,
+  onChange,
+}: {
+  graphDatasourceUid?: string;
+  query: TempoQuery;
+  onChange: (value: TempoQuery) => void;
+}) {
+  const dsState = useAsync(() => getDS(graphDatasourceUid), [graphDatasourceUid]);
+  if (dsState.loading) {
+    return null;
+  }
+
+  const ds = dsState.value as LokiDatasource;
```

Comment:
Should this be `as PrometheusDatasource`?

Reply:
Good catch yeah should be prometheus